### ISSUE TITLE
Troubles trying to use ParticleNativeAPI on mc version 1.7.10

### ISSUE THREAD

**Issue by HyperGamer11** [Fierioziy/ParticleNativeAPI#4]
Hello, I am new to github so I don't know how to use it well, I am having troubles with using this plugin on the 1.7.10 version of minecraft (My server is using Thermos-1.7.10-1558-server) and my server is giving me this stacktrace:

`[20:01:00 ERROR]: [ParticleNativeAPI] Failed to dynamically create class library using ASM.
[20:01:00 ERROR]: [ParticleNativeAPI] This API might not be compatible with current server version.
[20:01:00 ERROR]: [ParticleNativeAPI] If you suspect it is a bug, report it on issue tracker on plugin's github page with provided stacktrace:
[20:01:00 WARN]: com.github.fierioziy.particlenativeapi.api.utils.ParticleException: Failed to load particle library.
[20:01:00 WARN]:        at com.github.fierioziy.particlenativeapi.core.ParticleNativeCore.setupCore(ParticleNativeCore.java:67)
[20:01:00 WARN]:        at com.github.fierioziy.particlenativeapi.core.ParticleNativeCore.loadAPI(ParticleNativeCore.java:32)
[20:01:00 WARN]:        at com.github.fierioziy.particlenativeapi.plugin.ParticleNativePlugin.onLoad(ParticleNativePlugin.java:41)
[20:01:00 WARN]:        at org.bukkit.craftbukkit.v1_7_R4.CraftServer.loadPlugins(CraftServer.java:348)
[20:01:00 WARN]:        at net.minecraft.world.storage.SaveHandler.initBukkitData(SaveHandler.java:462)
[20:01:00 WARN]:        at net.minecraft.world.storage.SaveHandler.func_75757_d(SaveHandler.java:138)
[20:01:00 WARN]:        at net.minecraft.world.World.<init>(World.java:374)
[20:01:00 WARN]:        at net.minecraft.world.WorldServer.<init>(WorldServer.java:162)
[20:01:00 WARN]:        at net.minecraft.server.MinecraftServer.func_71247_a(MinecraftServer.java:375)
[20:01:00 WARN]:        at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:337)
[20:01:00 WARN]:        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:643)
[20:01:00 WARN]:        at java.lang.Thread.run(Unknown Source)
[20:01:00 WARN]: Caused by: com.github.fierioziy.particlenativeapi.api.utils.ParticleException: Error: this server version is not supported!
[20:01:00 WARN]:        at com.github.fierioziy.particlenativeapi.core.asm.ParticlesASM.<init>(ParticlesASM.java:65)
[20:01:00 WARN]:        at com.github.fierioziy.particlenativeapi.core.ParticleNativeCore.setupCore(ParticleNativeCore.java:55)
[20:01:00 WARN]:        ... 11 more`

I would like to know how to use this plugin for the version 1.7.10 because I am using the examples you gave on the spigot website and it isnt working.

**Comment by Fierioziy** [comment]
Hello, thanks for providing a stacktrace, it helped me locating a problem.

**TL;DR:** Thermos server strongly obfuscate minecraft internal classes, I can't even make a patch for it (probably any particle API couldn't do it as well).

**A little longer:**
Unfortunately, ParticleNativeAPI does not support Bukkit forks that significantly interferes with Minecraft internal classes (since this API is partially based on them).
As I've checked, **Thermos is one of those kind of fork, that have made strong obfuscation** (and changes) in Minecraft internal classes making API unable to locate some of them.
I'm pretty sure, that **any particle API won't be compatible with this server type**.

Bukkit and Spigot does not obfuscate classes (they even partially deobfuscate them!) and any server forks that is based on them is most likely compatible with ParticleNativeAPI.

**Comment by HyperGamer11** [comment]
Thanks for your help :D! I was really confused beacuse none of the particle plugins that I've tested worked.

**Comment by Fierioziy** [comment]
Nice to hear :) I'll try to check if there is possibility to work around Forge servers (or forks), but at first glance It will most likely not be possible.

**Comment by Fierioziy** [comment]
Hi, I'm closing it in favor of [this issue](https://github.com/Fierioziy/ParticleNativeAPI/issues/6) which is very related to this issue.

I suggest subscribing to notifications on this linked issue for further updates if you are interested in it
